subprojects/core/src/main/java/org/gradle/process/internal/JvmForkOptions.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process.internal;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.process.JavaForkOptions;
+
+import java.io.File;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.gradle.process.internal.util.MergeOptionsUtil.containsAll;
+import static org.gradle.process.internal.util.MergeOptionsUtil.getHeapSizeMb;
+import static org.gradle.process.internal.util.MergeOptionsUtil.normalized;
+
+/**
+ * Represents effective options for forking a Java process.
+ *
+ * Strongly relates to {@link JavaForkOptions}.
+ */
+@NonNullApi
+public class JvmForkOptions {
```

Comment:
I wonder if this class could get a better name. I'm not particularly committed to any of these, but here are some ideas:
- `EffectiveJavaForkOptions`
- `ComputedJavaForkOptions` (rename `toEffectiveJvmForkOptions()` to `computeEffectiveJavaForkOptions` for additional clarity?)
- Previous idea, but with `Baked`.

As-is, it's confusing at first glance what the difference is between the two types, and it could be confusing when debugging.

Reply:
I think that makes sense. I think `EffectiveJavaForkOptions` tells me the most, what this class represents, although I don't like the length of the name, but I think we can live we that for now.

Reply:
Done

Reply:
Done